### PR TITLE
feat(src): support object merging and recursive variable subst

### PR DESCRIPTION
**Pull request by z8674558**

### CODE REVIEW

**Comment by zmstone** [comment]
Please add test cases in `test/hocon_test.erl`.
files in `sample-configs` dir is a collection from  https://github.com/lightbend/config

File: src/hocon.erl [hunk 1]
```diff
@@ -107,7 +107,7 @@ parse(Tokens) ->
 
 -spec(transform(config(), ctx()) -> {ok, config()}).
 transform(Config, Ctx) ->
-    try include(substitute(Config), Ctx) of
+    try include(Config, Ctx) of
```
Comment (zmstone) [inline src/hocon.erl:110]: missing substitute ?

Comment (z8674558) [reply src/hocon.erl:110]: substitution is now done by resolve/1

Comment (z8674558) [reply src/hocon.erl:110]: which means that substitution occurs before the kv-list is converted into map.

**Comment by z8674558** [comment]
Closing in favor of https://github.com/emqx/hocon/pull/22

**Pull request closed without merging**